cmake/modules/AddSwift.cmake

```diff
@@ -536,8 +535,29 @@ function(add_swift_host_library name)
         "LINKER:-current_version,${SWIFT_COMPILER_VERSION}")
     endif()
 
+    # Always link as CXX even if we have swift content unless we only contain
+    # swift content signaled via us being marked "PURE_SWIFT".
+    if (NOT ASHL_PURE_SWIFT)
+      set_target_properties(${name} PROPERTIES LINKER_LANGUAGE CXX)
+
+      # If we found a swift compiler and are going to use swift code in swift
+      # host side tools but link with clang, add the appropriate -L paths so we
+      # find all of the necessary swift libraries.
+      if (CMAKE_Swift_COMPILER)
+        # Add in the SDK directory for the host platform and add an rpath.
+        target_link_libraries(${name} PRIVATE
+          -L${SWIFT_SDK_${SWIFT_HOST_VARIANT_SDK}_ARCH_${SWIFT_HOST_VARIANT_ARCH}_PATH}/usr/lib/swift)
+        # Add in the toolchain directory so we can grab compatibility libraries
+        get_filename_component(TOOLCHAIN_BIN_DIR ${CMAKE_Swift_COMPILER} DIRECTORY)
+        get_filename_component(TOOLCHAIN_LIB_DIR "${TOOLCHAIN_BIN_DIR}/../lib/swift/macosx" ABSOLUTE)
+        target_link_libraries(${name} PUBLIC
```

Comment:
This should use `target_link_directories` and drop the `-L` in the option.

Reply:
I'm fine with that.

Reply:
(this is the important part to fix IMO)